File: BOJ/#2581 소수/소수.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int M,N; cin>>M>>N;

    // 에라토스테네스의 체
    vector<bool> isPrime(N+1, true);
    isPrime[1] = false;

    for(int i=3; i<=N; i++) {
        if(!(i%2)) isPrime[i] = false;
        else if(isPrime[i])
            for(int j=i*2; j<=N; j+=i) isPrime[j] = false;
    }

    int sum=0, minPrime;
    for(int i=N; i>=M; i--) {
        if(isPrime[i]) {
            sum += i;
            minPrime = i;
        }
    }

    if(!sum) cout<<-1;
    else cout<<sum<<endl<<minPrime;
}

File: BOJ/#2581 소수/소수.py
M = int(input())
N = int(input())

isPrime = [True for i in range(N+1)]
isPrime[1] = False

for i in range(3, N+1):
    if i%2 == 0: isPrime[i] = False
    elif isPrime[i]:
        for j in range(i*2, N+1, i): isPrime[j] = False

sum = 0
for i in range(N, M-1, -1):
    if isPrime[i]:
        sum += i
        minPrime = i

if sum == 0: print(-1)
else: print(sum, minPrime, sep='\n')
